### DISCUSSION TITLE
start_preview() fails when cv2 is imported

### DISCUSSION THREAD

**Discussion by carterw** [raspberrypi/picamera2#763] [Q&A] [answered]
Sorry if this has been answered elsewhere, but I'm trying to run a simple picamera2 program with a preview pane and cv2. I'm finding that the program runs just fine without cv2, but if I so much as import cv2 the program crashes at the picam2.start_preview(Preview.QTGL) statement. I'm running Bullseye with recent updates. Do I have some kind of version mismatch?
I have cv2 version 4.7.0 installed. 

The exception is;
`QObject::moveToThread: Current thread (0x7f8001afe0) is not the object's thread (0x7f80225e50).
Cannot move to target thread (0x7f8001afe0)QObject::moveToThread: Current thread (0x7f8001afe0) is not the object's thread (0x7f80225e50).
Cannot move to target thread (0x7f8001afe0)`

Here's the program;

import cv2 # fails if I import this

from picamera2 import Picamera2, Preview
import time

picam2 = Picamera2()
camera_config = picam2.create_still_configuration(
    main={"format": 'XRGB8888', "size": (1920, 1080)}, 
    lores={"size": (640, 480)}, 
    display="main")
picam2.configure(camera_config)
picam2.start_preview(Preview.QTGL)  # exception happens here
picam2.start()
time.sleep(2)
picam2.capture_file("test.jpg")

**Comment by carterw** [comment, marked as answer]
It turned out to be the openCV version, apparently I jumped the gun when I installed that. A little too green?

**Reply by carterw** [reply]
A little more info on this. It turned out that originally I had used pip to install cv2;
`pip install opencv-python`

If you do that you get a cv2 version that is north of 4.7.0, and for me that crashed when start_preview() was invoked. I used pip to uninstall that, and reinstalled cv2 with;
`sudo apt install python3-opencv`

This gives a cv2 version in the range of 4.5, which does not crash with start_preview(). 

So I don't know what change was made in recent versions of cv2, probably having something to do with the threading, but it was a breaking one in this case.

**Comment by knelson-hme** [comment]
I had this same problem and solved it by uninstalling the "normal" python opencv `python -m pip uninstall opencv-python` and installing the headless version of opencv `python -m pip install opencv-python-headless`
